cbits/ctermid.c

```diff
@@ -0,0 +1,21 @@
+/* Provide ctermid() if it is missing (as on some Android versions). */
+
+#include "HsUnixConfig.h"
+#ifndef HAVE_CTERMID
+
+#include "ghcplatform.h"
+
+#if defined(linux_HOST_OS) || defined(linux_android_HOST_OS)
```

Comment:
Because the implementation is only necessarily correct on Linux systems, where `/dev/tty` is the name of the controlling terminal. On other exotic systems, perhaps the name is something else. What feature would you like to detect with Autoconf for this, while remaining friendly to cross-compilation?